libuuid/src/gen_uuid.c

```diff
@@ -250,7 +255,8 @@ static int get_clock(uint32_t *clock_high, uint32_t *clock_low,
 	}
 	if (state_fd >= 0) {
 		rewind(state_f);
-		while (flock(state_fd, LOCK_EX) < 0) {
+		lock.l_type = F_WRLCK;
+		while (fcntl(state_fd, F_SETLKW, &lock) < 0) {
 			if ((errno == EAGAIN) || (errno == EINTR))
```

Comment:
Well... "git blame" is your friend :-) You'll ses we had fcntl() there and it was removed by commit c190c6c2e987816ca0f273e2de528701d10cbbc1.

It would be better to introduce lock_state() and unlock_state() static functions and use 
```
#ifdef HAVE_FLOCK
   ... lock by flock() ...
#else
   ... lock by fcntl()...
#endif